### PR TITLE
[RW-7398][risk=low] Add enforce login.gov ff

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/utils/access-utils.tsx [hunk 1]
```diff
@@ -90,62 +90,62 @@ interface AccessModuleConfig {
 // without a next step:
 // https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java#L240-L272
 export const getAccessModuleConfig = (moduleName: AccessModule): AccessModuleConfig => {
-  const {enableRasLoginGovLinking, enableEraCommons, enableComplianceTraining} = serverConfigStore.get().config;
```
Comment (yonghaoy) [inline ui/src/app/utils/access-utils.tsx:93]: I think the indent size should be 2 per: 
https://github.com/all-of-us/workbench/blob/master/ui/intellij-typescript-style.xml
But let me know that I should revert this.

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java [hunk 2]
```diff
@@ -224,7 +224,7 @@ private boolean isModuleEnabledInEnvironment(AccessModule module) {
       case COMPLIANCE_TRAINING:
         return accessConfig.enableComplianceTraining;
       case RAS_LINK_LOGIN_GOV:
-        return accessConfig.enableRasLoginGovLinking;
+        return accessConfig.enforceRasLoginGovLinking || accessConfig.enableRasLoginGovLinking;
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:227]: I'm not sure we should change this

File: ui/src/app/pages/admin/admin-institution-edit.tsx [hunk 3]
```diff
@@ -123,7 +123,7 @@ const nonEmpty = (item: string): boolean => item && !!item.trim();
 
 const EraRequiredSwitch = (props: {tierConfig: InstitutionTierConfig, onChange: (boolean) => void}) => {
   const {tierConfig, onChange} = props;
-  const {config: {enableRasLoginGovLinking}} = useStore(serverConfigStore);
+  const {config: {enableRasLoginGovLinking, enforceRasLoginGovLinking}} = useStore(serverConfigStore);
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.tsx:126]: this is not used

File: ui/src/app/pages/homepage/data-access-requirements.spec.tsx [hunk 4]
```diff
@@ -50,11 +50,17 @@ describe('DataAccessRequirements', () => {
     });
 
     it('should not return the RAS module from getEnabledModules when its feature flag is disabled', () => {
-        serverConfigStore.set({config: {...defaultServerConfig, enableRasLoginGovLinking: false}});
+        serverConfigStore.set({config: {...defaultServerConfig, enableRasLoginGovLinking: false, enforceRasLoginGovLinking: false}});
         const enabledModules = getEnabledModules(allModules);
         expect(enabledModules.includes(AccessModule.RASLINKLOGINGOV)).toBeFalsy();
     });
 
+    it('should return the RAS module from getEnabledModules when enforceRasLoginGovLinking is enabled, enableRasLoginGovLinking is not', () => {
```
Comment (jmthibault79) [inline ui/src/app/pages/homepage/data-access-requirements.spec.tsx:58]: when would this combination make sense?

File: ui/src/app/utils/access-utils.tsx (hunk 1 above)
Comment (jmthibault79) [reply ui/src/app/utils/access-utils.tsx:93]: I don't think we have a linter reliably enforcing this right now.  Until we fix that, I'm OK with updating spacing to fit your local environment.

Comment (yonghaoy) [reply ui/src/app/utils/access-utils.tsx:93]: I got that from my intellij: https://docs.google.com/document/d/1DtESBapEzvuti7xODTFPHorwmLM7LybF-6D5lhbIkLU/edit#

File: ui/src/app/pages/admin/admin-institution-edit.tsx (hunk 3 above)
Comment (yonghaoy) [reply ui/src/app/pages/admin/admin-institution-edit.tsx]: removed the unused one. Initially I want the toggle being controlled by enforce and enable. That means in the future, if RAS is not enforced in one env, we will not allow institution switch this toggle. But now I feel that we should allow admin switch the toggle regardless RAS enforcement. 
That said, if RAS is not enabled, eRA can be required or not depends on institution needs. 
In in real world, that is only for preprod in which eRA and RAS both should be disabled. Showing this is fine I guess.

File: ui/src/app/utils/access-utils.tsx (hunk 1 above)
Comment (yonghaoy) [reply ui/src/app/utils/access-utils.tsx:93]: I think that doc is part of onboarding process when setting up local development

File: api/src/test/java/org/pmiops/workbench/access/UserServiceAccessTest.java [hunk 5]
```diff
@@ -1018,6 +1023,49 @@ public void testInstitutionRequirement_optionalEra_loginGovFlagEnabled_eRAFlagDi
...
+
+    // Complete RAS Linking, verify user become registered
+    accessModuleService.updateCompletionTime(
+        dbUser, AccessModuleName.RAS_LOGIN_GOV, new Timestamp(PROVIDED_CLOCK.millis()));
+    updateUserWithRetries(Function.identity());
+    assertRegisteredTierEnabled(dbUser);
+  }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
```
Comment (yonghaoy) [inline api/src/test/java/org/pmiops/workbench/access/UserServiceAccessTest.java:1047]: This is the cool Junit 5 feature @calbach presented last week, looks beautiful!

File: api/src/test/java/org/pmiops/workbench/access/UserServiceAccessTest.java [hunk 6]
```diff
@@ -1018,6 +1021,26 @@ public void testInstitutionRequirement_optionalEra_loginGovFlagEnabled_eRAFlagDi
     assertRegisteredTierEnabled(dbUser);
   }
 
+  @Test
+  public void testRasLinkNotComplete() {
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/test/java/org/pmiops/workbench/access/UserServiceAccessTest.java:1027]: sorry - let me clarify.  I want this test:
* enable = true
* enforce = false
* user does not have RAS completion
* assert(user is registered)

File: ui/src/app/pages/homepage/data-access-requirements.spec.tsx (hunk 4 above)
Comment (jmthibault79) [reply ui/src/app/pages/homepage/data-access-requirements.spec.tsx:58]: keep it

File: api/src/test/java/org/pmiops/workbench/access/UserServiceAccessTest.java [hunk 7]
```diff
@@ -1018,6 +1023,64 @@ public void testInstitutionRequirement_optionalEra_loginGovFlagEnabled_eRAFlagDi
...
+
+    // Complete RAS Linking, verify user become registered
+    accessModuleService.updateCompletionTime(
+        dbUser, AccessModuleName.RAS_LOGIN_GOV, new Timestamp(PROVIDED_CLOCK.millis()));
+    updateUserWithRetries(Function.identity());
+    assertRegisteredTierEnabled(dbUser);
+  }
+
+  @Test
+  public void testRasLinkNotComplete_enforceRasOff_enableRasOn() {
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/access/UserServiceAccessTest.java:1047]: 👍

File: ui/src/app/utils/access-utils.tsx (hunk 1 above)
Comment (jmthibault79) [reply ui/src/app/utils/access-utils.tsx:93]: Thanks!  I thought I had that configured but I must have lost it somehow.

**Pull request merged**